database/service_db.go

```diff
@@ -83,55 +63,49 @@ func (d *LevelDBServiceDB) All() ([]*service.Service, error) {
 }
 
 // Delete deletes service from database.
-func (d *LevelDBServiceDB) Delete(hash hash.Hash) error {
-	tx, err := d.db.OpenTransaction()
+func (d *ServiceDB) Delete(hash hash.Hash) error {
+	has, err := d.s.Has(hash)
 	if err != nil {
 		return err
 	}
-	if _, err := tx.Get(hash, nil); err != nil {
-		tx.Discard()
-		if err == leveldb.ErrNotFound {
-			return &ErrNotFound{resource: "service", hash: hash}
-		}
-		return err
+	if !has {
+		return &ErrNotFound{resource: "service", hash: hash}
```

Comment:
Also `ErrNotFound` should not be used anymore. A method like `Has` should be exposed.

The database.IsErrNotFound is used exactly **once** in instance sdk package:

```go
  // check if instance already exists
  _, err = i.instanceDB.Get(instanceHash)
  if err == nil {
    return nil, &AlreadyExistsError{Hash: instanceHash}
  } 
  if !database.IsErrNotFound(err) {
    return nil, err
  }
```

`Has` should be used here and only implemented in InstanceDB because there is needed.


So to summurize, here fmt.Error should be used + with refactoring instance db the ErrNotFound type should completly gone.

Reply:
Btw the same story is with `AlreadyExistsError` in service sdk.